server/src/main/antlr3/org/apache/iotdb/db/sql/parse/TqlParser.g

```diff
@@ -0,0 +1,726 @@
+        throws RecognitionException {
+    throw re;
+}
+
+@Override
+public void displayRecognitionError(String[] tokenNames, RecognitionException e) {
+    errors.add(new ParseError(this, e, tokenNames));
+}
+
+@Override
+public String getErrorHeader(RecognitionException e) {
+String header = null;
+if (e.charPositionInLine < 0 && input.LT(-1) != null) {
+    Token t = input.LT(-1);
+    header = "line " + t.getLine() + ":" + t.getCharPositionInLine();
+} else {
+    header = super.getErrorHeader(e);
+}
+return header;
+}
+
+@Override
+public String getErrorMessage(RecognitionException e, String[] tokenNames) {
+    String msg = null;
+    // Translate the token names to something that the user can understand
+    String[] tokens = new String[tokenNames.length];
+    for (int i = 0; i < tokenNames.length; ++i) {
+        tokens[i] = TqlParser.getTokenName(tokenNames[i]);
+    }
+
+    if (e instanceof NoViableAltException) {
+        @SuppressWarnings("unused")
+        NoViableAltException nvae = (NoViableAltException) e;
+        // for development, can add
+        // "decision=<<"+nvae.grammarDecisionDescription+">>"
+        // and "(decision="+nvae.decisionNumber+") and
+        // "state "+nvae.stateNumber
+        msg = "cannot recognize input near "
+            + input.LT(1) != null ? " " + getTokenErrorDisplay(input.LT(1)) : ""
+            + input.LT(1) != null ? " " + getTokenErrorDisplay(input.LT(1)) : ""
+            + input.LT(3) != null ? " " + getTokenErrorDisplay(input.LT(3)) : "";
+    } else if (e instanceof MismatchedTokenException) {
+        MismatchedTokenException mte = (MismatchedTokenException) e;
+        msg = super.getErrorMessage(e, tokens) + (input.LT(-1) == null ? "":" near '" + input.LT(-1).getText()) + "'"
+            + ". Please refer to SQL document and check if there is any keyword conflict.";
+    } else if (e instanceof FailedPredicateException) {
+        FailedPredicateException fpe = (FailedPredicateException) e;
+        msg = "Failed to recognize predicate '" + fpe.token.getText() + "'. Failed rule: '" + fpe.ruleName + "'";
+    } else {
+        if(tokenNameMap.containsKey("K_"+e.token.getText().toUpperCase())){
+        msg = e.token.getText() + " is a key word. Please refer to SQL document and check whether it can be used here or not.";
+    } else {
+        msg = super.getErrorMessage(e, tokens);
+    }
+}
+return messages.size() > 0 ? msg + " in " + messages.peek() : msg;
+}
+}
+
+@rulecatch {
+catch (RecognitionException e) {
+    reportError(e);
+    throw e;
+}
+}
+statement
+    : sqlStatement (SEMI)? EOF
+    ;
+
+sqlStatement
+    : ddlStatement
+    | dmlStatement
+    | administrationStatement
+    ;
+
+dmlStatement
+    : selectStatement
+    | insertStatement
+    | updateStatement
+    | deleteStatement
+    | loadStatement
+    ;
+
+ddlStatement
+    : createTimeseries
+    | deleteTimeseries
+    | setStorageGroup
+    | deleteStorageGroup // todo to implement
```

Comment:
It is implemented now.

Reply:
So far, the function is implemented. Thanks for your reminding.